server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -0,0 +1,363 @@
+     * @param clientID      client id
+     * @return properties for producer
+     */
+    private Properties setKafkaProducer(String serverAddress, String clientID) {
+
+        Properties p = new Properties();
+        p.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, serverAddress);
+        p.setProperty(ProducerConfig.CLIENT_ID_CONFIG, clientID);
+        p.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                StringSerializer.class.getName());
+        p.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                StringSerializer.class.getName());
+
+        return p;
+    }
+
+    /**
+     * Get host name of this consumer.
+     *
+     * @return consumers host name
+     */
+    private String getConsumerHostName() {
+        try {
+            return InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+            logger.error("Could not find the consumer's hostname.");
+        }
+        return "Unknown";
+    }
+
+    /**
+     * This is a utility method to get current committed offset of all partitions for a consumer.
+     *
+     * @param consumer Kafka consumer
+     * @param topics   list of topics
+     */
+    private void getOffsetForPartitions(KafkaConsumer<String, String> consumer,
+                                        List<String> topics) {
+
+        for (String t : topics) {
+            for (PartitionInfo p : consumer.partitionsFor(t)) {
+                emitMessage(this.serverLog, this.serverLogTopic,
+                        "T: " + t + " P: " + p.partition() + " OfC: "
+                                + consumer.committed(new TopicPartition(t, p.partition()))
+                                .offset());
+            }
+        }
+    }
+
+    /**
+     * This method adds one to the offset of all the partitions of a topic.
+     * This is useful when you want to skip an offset with FATAL errors when
+     * the FASTEN server is restarted.
+     * Please note that this is NOT the most efficient way to restart FASTEN server
+     * in the case of FATAL errors.
+     */
+    private void skipPartitionOffsets() {
+        ArrayList<TopicPartition> topicPartitions = new ArrayList<>();
+        List<String> topics = new ArrayList<>();
+        this.plugin.consumeTopics().ifPresentOrElse(topics::addAll, () -> {
+        });
+        if (topics.isEmpty()) {
+            return;
+        }
+        // Note that this assumes that the consumer is subscribed to one topic only
+        for (PartitionInfo p : this.connection.partitionsFor(topics.get(0))) {
+            topicPartitions.add(new TopicPartition(topics.get(0), p.partition()));
+        }
+
+        ConsumerRecords<String, String> records = dummyPoll(this.connection);
+
+        if (records.count() != 0) {
+            for (TopicPartition tp : topicPartitions) {
+                logger.debug("Topic: {} | Current offset for partition {}: {}", topics.get(0),
```

Comment:
Same as above: either log to Kafka or using the logger

Reply:
Switched everything to logger